zerver/tests/test_message_delete.py

```diff
@@ -48,7 +57,12 @@ def test_delete_message_by_admin(msg_id: int) -> "TestHttpResponse":
             result = self.client_delete(f"/json/messages/{msg_id}")
             return result
 
-        def test_delete_message_by_owner(msg_id: int) -> "TestHttpResponse":
+        def test_delete_message_by_moderator(msg_id: int) -> "TestHttpResponse":
+            self.login("shiva")
+            result = self.client_delete(f"/json/messages/{msg_id}")
+            return result
```

Comment:
Just a small note: These little helpers should be using `self.api_delete` instead of switching which user is logged in -- part of the principle of not mutating global state in little functions.

Might also be good to do a commit renaming them to avoid the `test_` prefix, which were these top-level functions, would mean tehy get executed as test methods. `try_delete_message_by_moderator` would be fine.

Maybe a good follow-up refactor commit.